engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/TypeOfBuiltin.java

```diff
@@ -13,12 +16,20 @@
     autoRegister = false)
 @GenerateUncached
 public final class TypeOfBuiltin extends Node {
+  private @Child WarningsLibrary warnings = WarningsLibrary.getFactory().createDispatched(11);
   private @Child TypeOfNode typeOf = TypeOfNode.create();
 
   private TypeOfBuiltin() {}
 
-  public Object execute(@AcceptsError Object value) {
-    return typeOf.execute(value);
+  public Object execute(@AcceptsError @AcceptsWarning Object value) {
+    if (warnings.hasWarnings(value)) {
+      try {
+        value = warnings.removeWarnings(value);
+      } catch (UnsupportedMessageException ex) {
+        // keep the old value
```

Comment:
Shouldn't here rather be `CompilerDirectives.shouldNotReachHere`? Or a Panic throw? If the receiver says that it `hasWarnings`, `removeWarnings` should not throw `UnsupportedMessageException`

Reply:
The whole code could be written like:
```java
      try {
        value = warnings.removeWarnings(value);
      } catch (UnsupportedMessageException ex) {
        // keep the old value
     }
```
e.g. without the `if`. Then it would do exactly what I want: _"remove warnings if value has them"_. Now the code does the same, it just adds `if` check - that's not bad and the code may be more readable, maybe efficient in interpreter.

Adding `CompilerDirectives.shouldNotReachHere` or a Panic throw wouldn't be testing my code, but the implementation of `warnings` library. I don't want to yield `TypeOfBuiltin` error when _some 3rd party_ code is broken. I've had these kind of issues with `InteropLibrary` before...

...however this is not `InteropLibrary` from 3rd party code, but our own library implementation from Enso - e.g. maybe you are right.

Reply:
Changed in fde1757fd1833b491623e68e80f6be4391e6e072